src/status_im2/contexts/shell/activity_center/events.cljs

```diff
@@ -124,7 +124,9 @@
 
 (rf/defn mark-as-read-success
   {:events [:activity-center.notifications/mark-as-read-success]}
-  [cofx notification]
+  [cofx {:keys [chat-id] :as notification}]
+  (when chat-id
+    (rf/dispatch [:chats-list/load-chat chat-id]))
```

Comment:
@qfrank we shouldn't make events stateful. Calling dispatch is either done inside effects like `:json-rpc/call` or in views.

The code should probably be like this (I roughly tested it). And you will need to update the unit test when using `reg-event-fx` instead of `rf/defn`.

```diff
modified   src/status_im2/contexts/shell/activity_center/events.cljs
@@ -122,12 +122,13 @@
                       :on-error   [:activity-center/process-notification-failure notification-id
                                    :notification/mark-as-read]}]}))
 
-(rf/defn mark-as-read-success
-  {:events [:activity-center.notifications/mark-as-read-success]}
-  [cofx {:keys [chat-id] :as notification}]
-  (when chat-id
-    (rf/dispatch [:chats-list/load-chat chat-id]))
-  (notifications-reconcile cofx [(assoc notification :read true)]))
+(defn mark-as-read-success
+  [_ [{:keys [chat-id] :as notification}]]
+  {:fx [(when chat-id
+          [:dispatch [:chats-list/load-chat chat-id]])
+        [:dispatch [:activity-center.notifications/reconcile [(assoc notification :read true)]]]]})
+
+(rf/reg-event-fx :activity-center.notifications/mark-as-read-success mark-as-read-success)
 
 (rf/defn mark-as-unread
   {:events [:activity-center.notifications/mark-as-unread]}
modified   src/status_im2/contexts/shell/activity_center/events_test.cljs
@@ -69,7 +69,7 @@
                   (fn [& args]
                     (reset! f-args args)
                     :result)]
-      (is (= :result (events/mark-as-read-success cofx notif)))
+      (is (= :result (events/mark-as-read-success cofx [notif])))
       (is (= [cofx [(assoc notif :read true)]]
              @f-args)))))
```

Notice here we use a feature from `:fx` explicitly documented by re-frame, where nil values are ignored, so we can conveniently & conditionally build the vector of effects to be dispatched in order.

If you want to keep using `rf/defn` here to be consistent with the namespace, you'd need to use `rf/merge`.

Reply:
thank you for your reviewing! updated according your comment 🙂 @ilmotta

Reply:
Thanks, PR approved already!